### ISSUE TITLE
Esprima error with ember-suave 0.1.5

### ISSUE THREAD

**Issue by brzpegasus** [dockyard/ember-suave#17]
The latest release appears to be broken. If you pull it into an Ember app and try to build or run tests, you'll get the following error:

```
$ ember test
version: 0.2.3
Build failed.
File: app.js
Unknown node type ExportDefaultDeclaration.
Error: Unknown node type ExportDefaultDeclaration.
    at Controller.traverse (/Users/brzpegasus/Projects/foo/node_modules/ember-suave/node_modules/broccoli-jscs/node_modules/jscs/node_modules/estraverse/estraverse.js:517:31)
    at Object.traverse (/Users/brzpegasus/Projects/foo/node_modules/ember-suave/node_modules/broccoli-jscs/node_modules/jscs/node_modules/estraverse/estraverse.js:709:27)
    at Object.iterate (/Users/brzpegasus/Projects/foo/node_modules/ember-suave/node_modules/broccoli-jscs/node_modules/jscs/lib/tree-iterator.js:9:20)
    at Object.JsFile.iterate (/Users/brzpegasus/Projects/foo/node_modules/ember-suave/node_modules/broccoli-jscs/node_modules/jscs/lib/js-file.js:356:29)
    at Object.JsFile._buildNodeIndex (/Users/brzpegasus/Projects/foo/node_modules/ember-suave/node_modules/broccoli-jscs/node_modules/jscs/lib/js-file.js:730:14)
    at Object.JsFile (/Users/brzpegasus/Projects/foo/node_modules/ember-suave/node_modules/broccoli-jscs/node_modules/jscs/lib/js-file.js:39:24)
    at StringChecker._createJsFileInstance (/Users/brzpegasus/Projects/foo/node_modules/ember-suave/node_modules/broccoli-jscs/node_modules/jscs/lib/string-checker.js:187:16)
    at StringChecker.checkString (/Users/brzpegasus/Projects/foo/node_modules/ember-suave/node_modules/broccoli-jscs/node_modules/jscs/lib/string-checker.js:98:25)
    at JSCSFilter.processString (/Users/brzpegasus/Projects/foo/node_modules/ember-suave/node_modules/broccoli-jscs/index.js:58:31)
    at Filter.processFile (/Users/brzpegasus/Projects/foo/node_modules/ember-suave/node_modules/broccoli-jscs/node_modules/broccoli-filter/index.js:139:31)
```

**Comment by rwjblue** [comment]
Ok, so looks like the estraverse that is embedded with JSCS has issues with latest esprima. Will get this fixed ASAP.

**Comment by rwjblue** [comment]
Updating to estravers@4.0.0 fixes the `ExportDefaultSpecifier` bug you referenced above.

Another issue cropped up in an example application with spread arguments though:

```javascript
callSomeFunction(...someArray);
```

```bash
Error: Line 1: Unexpected token ...
```

**Comment by rwjblue** [comment]
It appears that the JSCS folks have added spread support to their custom version of [Esprima](https://github.com/jscs-dev/esprima-harmony), but the original issue is that they don't support destructuring with `get`/`set`.

I suppose we can revert https://github.com/dockyard/ember-suave/pull/16, to get us at least back to where we were, but the whole thing is annoying. :crying_cat_face:

**Comment by rwjblue** [comment]
So the question is: do we revert support for:

1. `let { get } = Ember;`
2. `someFunc(...someArray);`

**Comment by rwjblue** [comment]
Chatted with @brzpegasus, and we are going to revert #16 to get us back to the support we had in v0.1.4.

**Comment by brzpegasus** [comment]
According to Esprima's [ES6/2015 support roadmap](https://github.com/jquery/esprima/issues/1099), the spread operator won't be any time soon (it's slated for 2.x). Given their past release schedule, that could still be _at least_ a couple months away.

From [JSCS#1293](https://github.com/jscs-dev/node-jscs/issues/1293#issuecomment-96381940), I would guess there is no intention of moving to the latest Esprima while also breaking support for spreading, so we should be fine to continue using it.

Long story short, yes, let's go back to the functionality provided in 0.1.4. We can keep support for 1) `someFunc(...someArray)` and use the syntax workaround for 2), so `const { get: get } = Ember` or just `const get = Ember.get`;

**Comment by rwjblue** [comment]
Published fix as v0.1.6.

**Comment by zxqfox** [comment]
Just FYI: 2.x means "doesn't planned", but here it is: https://github.com/jscs-dev/node-jscs/issues/1293#issuecomment-96427422

**Comment by hzoo** [comment]
Ok released initial version of `babel-jscs` https://github.com/jscs-dev/babel-jscs/releases/tag/v1.0.1 which should support `let { get } = Ember;`.

Haven't tested it extensively though - also `validateQuoteMarks` doesn't work at the moment + need to use master .

**Comment by brzpegasus** [comment]
@hzoo Woo! Thanks, will check it out. :fireworks: :confetti_ball: :sparkles:
